tools/distpartitioning/gloo_wrapper.py

```diff
@@ -59,6 +59,74 @@ def alltoall_cpu(rank, world_size, output_tensor_list, input_tensor_list):
     for i in range(world_size):
         dist.scatter(output_tensor_list[i], input_tensor_list if i == rank else [], src=i)
 
+def alltoallv_cpu_data(rank, world_size, input_tensor_list, val_dtype):
+    """
+    Wrapper function to providing the alltoallv functionality by using underlying alltoall
+    messaging primitive
+
+    Parameters:
+    -----------
+    rank : int
+        The rank of current worker
+    world_size : int
+        The size of the entire
+    input_tensor_list : List of tensor
+        The tensors to exchange
+    val_dtype : torch dtype
+        torch dtype which is used to create buffer for receiving messages
+
+    Returns:
+    --------
+    list : 
+        list of tensors received from other processes during alltoall message
+
+    """
+    sizes = [list(x.size()) for x in input_tensor_list]
+    for idx in range(1,len(sizes)):
+        assert len(sizes[idx-1]) == len(sizes[idx])
+
+    #decide how much to pad. 
+    #always use the first-dimension for padding. 
+    ll = [ x[0] for x in sizes ]
+
+    #dims of the outgoing messages
+    out_dims = [ [np.amax(ll)] + l[1:] for idx, l in enumerate(sizes) ]
```

Comment:
why do you use `enumerate` here?

Reply:
Agreed. enumerate is not needed here since idx is not used in this instance. 
Changed the code appropriately.